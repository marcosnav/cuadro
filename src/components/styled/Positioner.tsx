import styled from 'styled-components';

interface IProps {
  bottom?: number;
  left?: number;
  moveX?: number;
  moveY?: number;
  right?: number;
  top?: number;
}

const position = (pos: number = 0) => {
  return pos + 'px';
};

const translate = (x: number = 0, y: number = 0) => {
  return `translate(${x}px, ${y}px)`;
};

const Positioner = styled.div<IProps>`
  bottom: ${({ bottom }) => position(bottom)};
  left: ${({ left }) => position(left)};
  position: fixed;
  right: ${({ right }) => position(right)};
  top: ${({ top }) => position(top)};
  transform: ${({ moveX, moveY }) => translate(moveX, moveY)};
  transition: all 0.6s ease;
`;

Positioner.displayName = 'Positioner';

export default Positioner;
