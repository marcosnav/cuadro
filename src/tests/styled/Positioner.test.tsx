import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import Positioner from './../../components/styled/Positioner';

describe('<Positioner/>', () => {
  const component = mount(<Positioner />);

  test('renders properly', () => {
    expect(component).toHaveStyleRule('bottom', '0px');
    expect(component).toHaveStyleRule('left', '0px');
    expect(component).toHaveStyleRule('right', '0px');
    expect(component).toHaveStyleRule('top', '0px');
    expect(component).toMatchSnapshot();
  });

  test('sets position and translates accordingly', () => {
    component.setProps({ bottom: 10, left: 15, right: 15, top: 45});
    expect(component).toHaveStyleRule('bottom', '10px');
    expect(component).toHaveStyleRule('left', '15px');
    expect(component).toHaveStyleRule('right', '15px');
    expect(component).toHaveStyleRule('top', '45px');

    component.setProps({ moveX: 100, moveY: 200 });
    expect(component).toHaveStyleRule('transform', 'translate(100px,200px)');
  });
});
