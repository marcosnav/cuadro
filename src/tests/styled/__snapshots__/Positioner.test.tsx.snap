// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Positioner/> renders properly 1`] = `
.c0 {
  bottom: 0px;
  left: 0px;
  position: fixed;
  right: 0px;
  top: 0px;
  -webkit-transform: translate(0px,0px);
  -ms-transform: translate(0px,0px);
  transform: translate(0px,0px);
  -webkit-transition: all 0.6s ease;
  transition: all 0.6s ease;
}

<Positioner>
  <div
    className="c0"
  />
</Positioner>
`;
